railties/lib/rails/generators.rb

```diff
@@ -269,6 +275,7 @@ def invoke(namespace, args = ARGV, config = {})
         if klass = find_by_namespace(names.pop, names.any? && names.join(":"))
           args << "--help" if args.empty? && klass.arguments.any?(&:required?)
           klass.start(args, config)
+          run_after_generate_callback if config[:behavior] == :invoke
```

Comment:
Does `start` return a value that we need to pass on? E.g. should we do `start(args, config).tap do`?

Reply:
The `start` returns all result of `all_command`.  `all_command` includes methods that do not relate with generating files(e.g. `check_class_collision`, `hook_for`). It is a little difficult to extract that only result of generating files.